feat: add 1.0.0 migration guide (#4776)
adobe/spectrum-web-components: projects/documentation/content/migrations/2024-10-01 (1.0.0).md

---
layout: guide.njk
title: 'Migration Guide: Spectrum Web Components (v1.0.0)'
displayName: 2024/10/01 (v1.0.0)
slug: migration-guide
---

> ## Migration: 2024/10/01 (v1.0.0)

Author: This date will get updated according to our timeline.

---
layout: guide.njk
title: 'Migration Guide: Spectrum Web Components (v1.0.0)'
displayName: 2024/10/01 (v1.0.0)
slug: migration-guide
---

## Migration: 2024/10/01 (v1.0.0)

> With the release of Spectrum Web Components version 1.0.0, we removed previously deprecated components and features.
> Starting with this version, Spectrum Web Components adheres to [semantic versioning (semver)](https://semver.org/), ensuring that any future breaking changes will be introduced only in major version updates. These changes will be communicated in advance, allowing developers to prepare and avoid disruption to existing applications. Patch releases, as part of this versioning approach, will be non-breaking and focus on fixes and minor improvements.

Reviewer: ```suggestion
With the release of Spectrum Web Components v1.0.0, we've removed previously deprecated components and features.

Starting with this version, Spectrum Web Components will adhere to [semantic versioning (semver)](https://semver.org/), which ensures that any future breaking changes are introduced only in major version bumps. These changes will be communicated in advance so that developers can prepare for the updates. As part of this versioning approach, patch releases will be non-breaking and will focus on fixes and minor improvements.
```

Minor tweaks

### 2. Quick actions (`sp-quick-actions`)

The `sp-quick-actions` component has been removed.
**Action Required**: If you are using `sp-quick-actions`, remove it from your application and replace it with the [action bar](/components/alert-banner/) component to allow users to perform actions on either a single or multiple items at the same time.

### 3. Split button (`sp-split-button`)

The `sp-split-button` component has been removed.
**Action Required**: If you are using `sp-split-button`, remove it from your application and replace it with the [button group](/components/button-group/) component to to show any additional actions related to the most critical action.

### 4. Action button (`sp-action-button`)

The `variant` attribute for `sp-action-button` has been removed.
**Action Required**: If your application uses the `variant` attribute, replace it with the `staticColor` attribute while keeping the same values. See the example below:

```ts
// Before
<sp-action-button variant="black"></sp-action-button>

// After

> <sp-action-button  staticColor="black"></sp-action-button>

Reviewer: Not necessary to address in this PR, obviously, but I'm surprised that the syntax is `staticColor` and not `static-color`, since it seems out of line with our other APIs.

Author: This is the code that should be written in the application source code. You are correct though, in the DOM this will become a `static-color` attribute.

Author: I updated these and the `overBackground` examples :) Thank you for the call-out!

```

### 5. Badge (`sp-badge`)

> The following deprecated fixed values for the `position` attribute on `sp-badge` have been removed: `top`, `bottom`, `left`, `right`.

Reviewer: ```suggestion
The values for the `position` attribute on `sp-badge` have been removed: `top`, `bottom`, `left`, `right`.
```

Do we need to say "fixed values" or can we just say values? I'm fine with either, but just curious.

**Action Required**: Replace these fixed values with their logical equivalents:

-   `top` → `block-start`
-   `bottom` → `block-end`
-   `left` → `inline-start`
-   `right` → `inline-end`

### 6. Button (`sp-button`)

Several deprecated values for the `variant` attribute on `sp-button` have been removed, including:

-   `black` and `white`: Use `staticColor="black"` or `staticColor="white"` instead.

> -   `overBackground`: Use `staticColor="white"` with `treatment="outline"` instead.

Reviewer: Oh, with `overBackground` written the way it is here, maybe I'm wrong about `staticColor`.

Author: `overBackground` is also a property that gets reflected as `over-background` in the DOM.

Author: Fixed!

// After
<sp-progress-circle staticColor="white"></sp-progress-circle>
```

Make sure to update all instances of `sp-progress-circle` where `overBackground` is used.

### 10. Theme (`sp-theme`)

The `theme` attribute for `sp-theme` has been removed.

**Action Required**: Replace the `theme` attribute with the `system` attribute. This is a drop-in replacement. See the example below:

```ts
// Before
<sp-theme theme="light"></sp-theme>

// After
<sp-theme system="light"></sp-theme>
```

Ensure that all instances of the `theme` attribute in your application are updated to use `system` instead.

### 11. Thumbnail (`sp-thumbnail`)

The following deprecated size values for `sp-thumbnail` have been removed: `xxs`, `xs`, `s`, `m`, `l`.

**Action Required**: Replace these deprecated sizes with their corresponding numeric values:

-   `xxs` → `100`
-   `xs` → `300`
-   `s` → `500`
-   `m` → `700`
-   `l` → `900`

See the example below for the migration:

```ts
// Before
<sp-thumbnail size="xs"></sp-thumbnail>

// After
<sp-thumbnail size="300"></sp-thumbnail>
```

Make sure to update all instances of `sp-thumbnail` using the deprecated `size` values.

## Final Notes

> Please ensure your project is fully compatible with Spectrum Web Components 1.0.0 before upgrading. Test thoroughly and review any deprecated or removed functionality as outlined above. If you encounter any issues during migration, consult the documentation or reach out for support.

Reviewer: ```suggestion
Please ensure your project is fully compatible with Spectrum Web Components 1.0.0 before upgrading. Test thoroughly and review any deprecated or removed functionality as outlined above. If you encounter any issues during migration, consult the documentation or reach out to the team for support.
```
